internal/stack/serverless.go

```diff
@@ -0,0 +1,465 @@
+		StackVersion: options.StackVersion,
+	}
+
+	return s, nil
+}
+
+func createProjectName(options Options) string {
+	return fmt.Sprintf("elastic-package-test-%s", options.Profile.ProfileName)
+}
+
+func newServerlessProvider(profile *profile.Profile) (*serverlessProvider, error) {
+	client, err := serverless.NewClient()
+	if err != nil {
+		return nil, fmt.Errorf("can't create serverless provider: %w", err)
+	}
+
+	return &serverlessProvider{profile, client, nil, nil}, nil
+}
+
+func (sp *serverlessProvider) BootUp(options Options) error {
+	logger.Warn("Elastic Serverless provider is in technical preview")
+
+	config, err := LoadConfig(sp.profile)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	settings, err := getProjectSettings(options)
+	if err != nil {
+		return err
+	}
+
+	_, ok := allowedProjectTypes[settings.Type]
+	if !ok {
+		return fmt.Errorf("serverless project type not supported: %s", settings.Type)
+	}
+
+	var project *serverless.Project
+
+	project, err = sp.currentProject(config)
+	switch err {
+	default:
+		return err
+	case errProjectNotExist:
+		logger.Infof("Creating %s project: %q", settings.Type, settings.Name)
+		config, err = sp.createProject(settings, options, config)
+		if err != nil {
+			return fmt.Errorf("failed to create deployment: %w", err)
+		}
+
+		project, err = sp.currentProject(config)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve latest project created: %w", err)
+		}
+
+		err = sp.createClients(project)
+		if err != nil {
+			return err
+		}
+
+		logger.Infof("Creating agent policy")
+		err = project.CreateAgentPolicy(options.StackVersion, sp.kibanaClient)
+		if err != nil {
+			return fmt.Errorf("failed to create agent policy: %w", err)
+		}
+
+		// TODO: Ensuring a specific GeoIP database would make tests reproducible
+		// Currently geo ip files would be ignored when running pipeline tests
+		// logger.Infof("Replacing GeoIP databases")
+	case nil:
+		logger.Debugf("%s project existed: %s", project.Type, project.Name)
+		printUserConfig(options.Printer, config)
+	}
+
+	logger.Infof("Starting local agent")
+	err = sp.startLocalAgent(options, config)
+	if err != nil {
+		return fmt.Errorf("failed to start local agent: %w", err)
+	}
+
+	return nil
+}
+
+func (sp *serverlessProvider) composeProjectName() string {
+	return DockerComposeProjectName(sp.profile)
+}
+
+func (sp *serverlessProvider) localAgentComposeProject() (*compose.Project, error) {
+	composeFile := sp.profile.Path(profileStackPath, ServerlessComposeFile)
+	return compose.NewProject(sp.composeProjectName(), composeFile)
+}
+
+func (sp *serverlessProvider) startLocalAgent(options Options, config Config) error {
+	err := applyServerlessResources(sp.profile, options.StackVersion, config)
+	if err != nil {
+		return fmt.Errorf("could not initialize compose files for local agent: %w", err)
+	}
+
+	project, err := sp.localAgentComposeProject()
```

Comment:
Something fails when trying to get logs from this agent :thinking: 


Running tests for haproxy package:
```
2023/09/04 20:19:05 DEBUG run command: /usr/bin/docker cp elastic-package-stack-serverless-elastic-agent-1:/usr/share/elastic-agent/state/data/logs/ /tmp/test-system-3434672278/logs/elastic-agent-internal
Error: error running package system tests: could not complete test run: check log messages failed: open /tmp/test-system-3434672278/logs/elastic-agent.log: no such file or directory
```

Reply:
Tested locally with all the latest changes, and it works for me testing haproxy package (from integrations repository).

Commands used:
```shell
elastic-package stack up -v -d --version 8.9.0 --provider serverless
elastic-package stack status
elastic-package test -v
elastic-package stack down -v
```

Tested also with `--version 8.0.0` (minimum version in package manifest)